control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -301,6 +301,19 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}
 
 	originalHostedControlPlane := hostedControlPlane.DeepCopy()
+	// Always update status based on the current state of the world.
+	defer func() {
```

Comment:
I'm not a huge fan of defer and patching, how about we keep the existing structure and have these missing images to be called and patched explicitly in e.g. l768?

Reply:
updated